File: backend/src/main/java/com/eyeclinic/surgerycenter/exception/BusinessException.java
package com.eyeclinic.surgerycenter.exception;

import com.eyeclinic.surgerycenter.enums.ErrorCode;
import lombok.Getter;

@Getter
public class BusinessException extends RuntimeException {
    private final String code;
    private final String message;

    public BusinessException(ErrorCode errorCode) {
        super(errorCode.getMessage());
        this.code = errorCode.getCode();
        this.message = errorCode.getMessage();
    }

    public BusinessException(ErrorCode errorCode, String message) {
        super(message);
        this.code = errorCode.getCode();
        this.message = message;
    }
}

File: src/common/business-exception.ts
import { ErrorCode, ErrorMessage } from './error-code';

export class BusinessException extends Error {
  code: ErrorCode;
  message: string;

  constructor(code: ErrorCode, message?: string) {
    super(message || ErrorMessage[code]);
    this.code = code;
    this.message = message || ErrorMessage[code];
    this.name = 'BusinessException';
  }
}
